tests/run-make/bare-outfile/rmake.rs

```diff
@@ -0,0 +1,15 @@
+// This test checks that manually setting the output file as a bare file with no file extension
+// still results in successful compilation.
+
+//@ ignore-cross-compile
+
+use run_make_support::{run, rustc, tmp_dir};
+use std::env;
+use std::fs;
+
+fn main() {
+    fs::copy("foo.rs", tmp_dir()).unwrap();
+    env::set_current_dir(tmp_dir());
+    rustc().output("foo").input("foo.rs");
```

Comment:
Suggestion: I think this is cleaner if it's written with `-o foo` + `--out-dir tmp_dir()`

> The output filename can be set with the [-o flag](https://doc.rust-lang.org/rustc/command-line-arguments.html#option-o-output). A suffix may be added to the filename with the [-C extra-filename flag](https://doc.rust-lang.org/rustc/codegen-options/index.html#extra-filename). The files are written to the current directory unless the [--out-dir flag](https://doc.rust-lang.org/rustc/command-line-arguments.html#option-out-dir) is used.

That way, we don't need the explicit copy of `foo.rs` at all.

i.e. `rustc().out_dir("foo").output("foo")`, depending on what you think about the name of `output` -- should it be renamed `output_filename`?

Reply:
I just realized that `rustc()` will always pass through this function (I think you wrote it). So, logically, we wouldn't need to specify the `--out-dir tmp_dir()`, as it's already set? 

I'm not too sure I understand your suggestion. In `rustc().out_dir("foo").output("foo")`, `.out_dir("foo")` is setting the file name `foo` as the output directory, but it's a filename, not a directory. I could certainly rename `output` to `output_filename`, but that would require replacing every instance of it in already ported tests.

```rs
fn setup_common() -> Command {
    let rustc = env::var("RUSTC").unwrap();
    let mut cmd = Command::new(rustc);
    set_host_rpath(&mut cmd);
    cmd.arg("--out-dir").arg(tmp_dir()).arg("-L").arg(tmp_dir());
    cmd
}
```

Reply:
My bad, I forgot by default the `Rustc` instance returned by `rustc()` we already set `--out-dir` to `tmp_dir()`, so `env::set_current_dir` and `fs::copy` etc. wasn't needed in the first place I think?

Reply:
And yeah, ignore the `output_filename` comment, I was just thinking if `output()` and `out_dir()` can become confusing.

Reply:
> My bad, I forgot by default the `Rustc` instance returned by `rustc()` we already set `--out-dir` to `tmp_dir()`, so `env::set_current_dir` and `fs::copy` etc. wasn't needed in the first place I think?

I will remove those 2 commands and see if that causes CI to fail.